crates/cheatnet/src/state.rs

```diff
@@ -129,6 +130,8 @@ pub enum CheatStatus<T> {
 #[derive(Clone)]
 pub struct CallTrace {
     pub entry_point: CallEntryPoint,
+    // Used to ensure on type level that we always have a class hash info
+    pub class_hash: ClassHash,
```

Comment:
It is `None` in `CallEntryPoint` if it can be deduced from the contract address - basically always `None` except library calls

Reply:
Resigning from this idea, bcs for failed calls we may not have it anyways (e.g. calling non existing contract address)